src/devices/machine/mm5740.cpp

```diff
@@ -110,22 +112,21 @@ TIMER_CALLBACK_MEMBER(mm5740_device::perform_scan)
 
 		for (int y = 0; y < 10; y++)
 		{
-			if (BIT(data, y))
-			{
-				uint8_t *rom = m_rom->base();
-				uint16_t offset = x*10 + y;
-				// Common portion
-				uint16_t common = (uint16_t) rom[offset];
+			u8 *rom = m_rom->base();
+			u16 offset = x * 10 + y;
+			// Common portion
+			u16 common = (u16)rom[offset];
 
-				offset += (((m_read_shift() ? 1: 0) + (m_read_control() ? 2: 0)) + 1) * 90;
+			offset += (((m_read_shift() ? 1 : 0) + (m_read_control() ? 2 : 0)) + 1) * 90;
 
-				// Unique portion based on shift/ctrl keys.
-				uint8_t uniq = rom[offset];
+			// Unique portion based on shift/ctrl keys.
+			u8 uniq = rom[offset];
 
-				uint16_t b = (((common & 0x10) << 4) | ((uniq & 0x0f) << 4) | (common & 0x0f)) ^ 0x1ff;
-
-				ako = true;
+			u16 b = (((common & 0x10) << 4) | ((uniq & 0x0f) << 4) | (common & 0x0f)) ^ 0x1ff;
 
+			if (BIT(data, y))
+			{
+				// Check for a new keypress
 				if (!BIT(m_x_mask[x], y))
 				{
 					m_x_mask[x] |= (1 << y);
```

Comment:
Only the calculation of `offset` is needed when a key is released, so the rest of these locals can stay inside the inner `if` block.

Reply:
But to compare offset with the offset used for the key down, I'll need a new class variable to store offset as calculated prior to it being updated with the shift/ctrl offset and the `* 90`. And it only should be saved when it passes all the if checks and save it at the same spot as the `m_b = b;` statement. Is that correct?